src/targets/gpu/include/migraphx/gpu/compile_hip.hpp

```diff
@@ -35,6 +36,30 @@ namespace migraphx {
 inline namespace MIGRAPHX_INLINE_NS {
 namespace gpu {
 
+struct hiprtc_src_file
+{
+    hiprtc_src_file() = default;
+    hiprtc_src_file(const src_file& s)
+        : path(s.path.string()), content(s.content.first, s.content.second)
+    {
+    }
+    std::string path;
+    std::string content;
+    operator src_file() const
+    {
+        return src_file{path, {content.data(), content.data() + content.size()}};
+    }
+    template <class Self, class F>
+    static auto reflect(Self& self, F f)
```

Comment:
Because `src_file` uses pointers to the source files which cannot be passed across process. So `hiprtc_src_file` is used instead with `std::string`. We already did this conversion in `hiprtc_program` constructor so I exposed it in the interface.